api/src/libs/migrateBusinessesFromUsers.ts

```diff
@@ -0,0 +1,53 @@
+import { BusinessesDataClient, UserDataClient } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { UserData } from "@shared/userData";
+
+export const migrateBusinessesFromUsers = async (
+  userDataClient: UserDataClient,
+  businessesDataClient: BusinessesDataClient,
+  logger: LogWriterType
+): Promise<void> => {
+  const processBatch = async (users: UserData[]): Promise<void> => {
+    for (const userData of users) {
+      const businessesData = userData.businesses;
+
+      if (businessesData && Object.keys(businessesData).length > 0) {
+        for (const businessId in businessesData) {
+          const businessData = businessesData[businessId];
+          const existingBusiness = await businessesDataClient.get(businessId);
+          const currentVersion = existingBusiness ? existingBusiness.version + 1 : 1;
+
+          await businessesDataClient.put({
+            ...businessData,
+            id: businessId,
+            version: currentVersion,
+          });
+
+          await userDataClient.put({
+            ...userData,
+            businesses: { [businessId]: { ...businessData, version: currentVersion } },
+          });
```

Comment:
This way we can keep track of which users have been migrated and which haven't. This would allow us to resume the migration in case we later need to resume the migration.

Reply:
@seidior We are currently incrementing  the migration/version number on line 18.

Reply:
I think what Spencer is trying to say is the one time migration should be v148 (currently v147 in UserData). The next migration will jump to v149. This way we can track who has migrated. This need to overlap with the Migrations function. 

Also, if my data is on version 99, this increments it to 100, and messes the migration because it's now skipping a migration (v99 to v100).

Reply:
Got what he was saying. I'm talkiing about the concept of the incrementing the versions. From the session yesterday we need to update the versions and reference the CURRENT_VERSION across both datastore/data objects.